### PR TITLE
[mempool] classify client-submitted vs. broadcasted transactions

### PR DESCRIPTION
**Pull request by bchocho**
### Description

Added a new label for classifying whether a mempool-to-commit latency was measured on a transaction that was directly submitted by the client to this node (via REST api) or came to this node via mempool broadcast.

### CODE REVIEW

File: mempool/src/core_mempool/transaction.rs [hunk 1]
```diff
@@ -84,6 +89,13 @@ pub struct SequenceInfo {
     pub account_sequence_number: u64,
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub struct InsertionInfo {
+    pub insertion_time: SystemTime,
+    pub client_submitted: bool,
+    pub is_end_to_end: bool,
```
Comment (JoshLind) [inline mempool/src/core_mempool/transaction.rs:96]: nit: can we add a comment to this line? I'm a noob and I'm not sure what `is_end_to_end` means? 🤔

**Review by JoshLind — APPROVED** [review]
Solid. Thanks @bchocho 😄

File: mempool/src/core_mempool/transaction.rs (hunk 1 above)
Comment (bchocho) [reply mempool/src/core_mempool/transaction.rs]: Good call. I renamed is_end_to_end and added a comment to both of those, as unfortunately they're quite nuanced.

**Review by igor-aptos — CHANGES_REQUESTED** [review]

File: mempool/src/core_mempool/transaction.rs [hunk 2]
```diff
@@ -84,6 +89,41 @@ pub struct SequenceInfo {
...
+    }
+
+    pub fn validator_scope_label(&self) -> &'static str {
+        if self.validator_end_to_end {
+            counters::E2E_LABEL
+        } else {
+            counters::LOCAL_LABEL
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (bchocho) [reply mempool/src/core_mempool/transaction.rs]: Thanks, made the suggested change to an enum. Please take a look!

File: mempool/src/core_mempool/transaction.rs [hunk 3]
```diff
@@ -84,6 +85,61 @@ pub struct SequenceInfo {
     pub account_sequence_number: u64,
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum SubmittedBy {
+    /// The transaction was received from a client REST API submission, rather than a mempool
+    /// broadcast. This can be used as the time a transaction first entered the network,
+    /// to measure end-to-end latency within the entire network. However, if a transaction is
+    /// submitted to multiple nodes (by the client) then the end-to-end latency measured will not
+    /// be accurate.
```
Comment (igor-aptos) [inline mempool/src/core_mempool/transaction.rs:94]: add in all comments after "will not be accurate" a comment ("will be lower than the correct value")

**Review by igor-aptos — APPROVED** [review]

**Pull request merged**